### ISSUE TITLE
Feature: Suggest a corrected PR title if determinable?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mtfoley/pr-compliance-action#367]
**Is your feature request related to a problem? Please describe.**

Sometimes users create PRs with titles that _almost_ match the conventional commit title. They'll do something like `Feature: Abc Def` instead of `feat: abc def`. Then they get confused by the complaint/error asking them to match the spec.

**Describe the solution you'd like**

For PR titles that match a known set of common mistakes such as that one, could we suggest the fix for them? Mocking up messaging:

> It looks like this PR's title could be edited to:
> ```
> feat: abc def
> ```

**Describe alternatives you've considered**

I suppose we could do it for them? But that's a bit more work. And I do want users to know to add correct PR titles in the first place...

**Comment by mtfoley** [comment]
To blow this out of proportion...

I saw a while back @bdougie had done a write up of commit message suggestions generated by AI (aicommits by @Nutlope). The package appears to have an optional to generate conventional commit messages, so it could be cool to do something with that:

https://dev.to/bdougieyo/ai-generated-git-commit-messages-4j7g
https://github.com/Nutlope/aicommits

The package is tooled as a CLI, so I'm not 100% how to use that with NCC-compiled things like this action. Also, there are some cases of mixed messaging that could come about, e.g. if you have a template for PR messages and at the same time, you kick back a generated suggestion that may have nothing to do with the template. Could be a feature issue worth filing on that repo 🤷 

And back down to what you mention here...
I think we could reuse a combination of techniques to piggy back the conventional commit parsing since it's got some pretty specific error messages - if they correctly specified a "subject" (e.g. "Feature: my nice things" yields `subject=Feature`, and the subject is in a dictionary of likely cases, say something useful), and then if there isn't a "subject" detected, try to compare the first word in the line to the same dictionary, provide a similarly helpful message.

**Comment by JoshuaKGoldberg** [comment]
Yeah I'm very hesitant to onboard AI in repository automation because of its nondeterminism:

* Even with very good safeguards to prevent vulgarity, it's a little risky
  * This concern is going to be less and less relevant over the years as AI improves, but AI is still new enough that I think the concern is pressing
* It's harder to debug when things go wrong, especially in the context of preventing them a next time

My preference would be to find or create a package that suggests a conventional commit name when given a non-conventional name based only on a set of rough heuristics. Being able to plug in an AI model would be a really nifty followup IMO.
